autoPyTorch/utils/pipeline.py

```diff
@@ -1,10 +1,13 @@
 # -*- encoding: utf-8 -*-
```

Comment:
I am thinking if the methods in this module would be better in the pipeline classes. E.g.:

- add a `self.task_type` attribute to the pipelines (maybe not even needed)
- the `BasePipeline` already implements `get_hyperparameter_search_space` (= the `get_configspace` function from here), it just needs the include/exclude support
- `get_dataset_requirements` is also already implemented in the `BasePipeline`
- create an util function `get_pipeline(task_type)` that returns the initialized pipeline. on that one can then call one of the above two methods

What do you think?

Reply:
To get dataset requirements we need to instantiate the pipeline, so the idea of having this in the utils file is that we use it to just get dataset_requirements and the config space for the general pipeline and the model_class attribute is used to generate models when trying to fit and predict a pipeline. Also, I think we need to send the dataset properties to get the pipeline configuration space as it is altered according to dataset properties. But I think it should be possible to instantiate the model class just to get the requirements and config space and hence we can avoid the double-checking of task type as being done right now.